src/main/java/net/sf/jabref/util/MassSetFieldAction.java

```diff
@@ -85,6 +87,18 @@ private void createDialog() {
         set = new JRadioButton(Localization.lang("Set fields"));
         rename = new JRadioButton(Localization.lang("Rename field to") + ":");
         rename.setToolTipText(Localization.lang("Move contents of a field into a field with a different name"));
+        
+        Collection<BibtexEntry> be = frame.basePanel().database().getEntries();
+        allFields = new TreeSet<String>();
+        for (BibtexEntry e : be) {
+            allFields.addAll(e.getAllFields());
+        }
```

Comment:
Can you extract the logic of getting all the fields of a database and move this method in the database class itself? Maybe use getAllVisibleFields as a method, then you can also ignore the ones that start with "__".